prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java

```diff
@@ -94,14 +94,16 @@
 
     //SUSPEND CHECKSTYLE CHECK ParameterNumberCheck
     @GET
-    @Path("list/{extension-name}")
+    @Path("list")
     @Produces({MediaType.TEXT_XML, MediaType.APPLICATION_JSON})
     public ExtensionJobList getExtensionJobs(
-            @PathParam("extension-name") String extensionName,
+            @QueryParam("extension-name") String extensionName,
```

Comment:
this API is used for two use case, one to get the jobs for an extension and other to get all jobs irrespective of registered extension. 
In the second case, we were not passing anything in extension-name, means null.. which resulted in error from FalconClient that the given parameter is null. So I changed it to QueryParam.